operators/multiclusterobservability/pkg/rendering/renderer_mcoa_test.go

```diff
@@ -169,12 +195,214 @@ func TestRenderAddonDeploymentConfig(t *testing.T) {
+				UserWorkloads: &mcov1beta2.UserWorkloadCapabilitiesSpec{
+					Logs: mcov1beta2.UserWorkloadLogsSpec{
+						Collection: mcov1beta2.UserWorkloadLogsCollectionSpec{
+							ClusterLogForwarder: mcov1beta2.ClusterLogForwarderSpec{
+								Enabled: true,
+							},
+						},
+					},
+				},
+			},
+			expectConfig: func(t *testing.T, cma *addonv1alpha1.ClusterManagementAddOn) {
+				assert.Len(t, cma.Spec.InstallStrategy.Placements[0].Configs, 0)
+			},
+		},
+		{
+			name: "add logs configs when both platform and user workloads are enabled",
+			capabilities: &mcov1beta2.CapabilitiesSpec{
+				Platform: &mcov1beta2.PlatformCapabilitiesSpec{
+					Logs: mcov1beta2.PlatformLogsSpec{
+						Collection: mcov1beta2.PlatformLogsCollectionSpec{
+							Enabled: true,
+						},
+					},
+				},
+				UserWorkloads: &mcov1beta2.UserWorkloadCapabilitiesSpec{
+					Logs: mcov1beta2.UserWorkloadLogsSpec{
+						Collection: mcov1beta2.UserWorkloadLogsCollectionSpec{
+							ClusterLogForwarder: mcov1beta2.ClusterLogForwarderSpec{
+								Enabled: true,
+							},
+						},
+					},
+				},
+			},
+			expectConfig: func(t *testing.T, cma *addonv1alpha1.ClusterManagementAddOn) {
+				assert.Len(t, cma.Spec.InstallStrategy.Placements[0].Configs, 1)
+			},
+		},
+		{
+			name: "add traces configs when user workloads is enabled",
+			capabilities: &mcov1beta2.CapabilitiesSpec{
+				UserWorkloads: &mcov1beta2.UserWorkloadCapabilitiesSpec{
+					Traces: mcov1beta2.UserWorkloadTracesSpec{
+						Collection: mcov1beta2.OpenTelemetryCollectionSpec{
+							Collector: mcov1beta2.OpenTelemetryCollectorSpec{
+								Enabled: true,
+							},
+							Instrumentation: mcov1beta2.InstrumentationSpec{
+								Enabled: true,
+							},
+						},
+					},
+				},
+			},
+			expectConfig: func(t *testing.T, cma *addonv1alpha1.ClusterManagementAddOn) {
+				assert.Len(t, cma.Spec.InstallStrategy.Placements[0].Configs, 2)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &MCORenderer{
+				cr: &mcov1beta2.MultiClusterObservability{
+					Spec: mcov1beta2.MultiClusterObservabilitySpec{
+						Capabilities: tt.capabilities,
+					},
+				},
+			}
+
+			// Add deployment config to reflect real input
+			baseCMA := &addonv1alpha1.ClusterManagementAddOn{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"k1": "v1",
+					},
+				},
+				Spec: addonv1alpha1.ClusterManagementAddOnSpec{
+					InstallStrategy: addonv1alpha1.InstallStrategy{
+						Placements: []addonv1alpha1.PlacementStrategy{
+							{
+								Configs: []addonv1alpha1.AddOnConfig{},
+							},
+						},
+					},
+				},
+			}
+			// to kustomize resource
+			baseCMAUnstructured, err := runtime.DefaultUnstructuredConverter.ToUnstructured(baseCMA)
+			assert.NoError(t, err)
+			kres := &kustomizeres.Resource{}
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(baseCMAUnstructured, kres)
+			assert.NoError(t, err)
+
```

Comment:
Why do it this way VS the way we did it for `TestRenderAddonDeploymentConfig`?

Reply:
What do you mean? I'm testing the ClusterManagementAddOn, not the AddOnDeploymentConfig.